test/e2e/util/mongotester/mongotester.go

```diff
@@ -432,7 +403,16 @@ func WithHosts(hosts []string) OptionApplier {
 }
 
 // WithTls configures the client to use tls
-func WithTls() OptionApplier {
+func WithTls(mdb mdbv1.MongoDBCommunity) OptionApplier {
+	if !mdb.Spec.Security.TLS.Enabled {
```

Comment:
one thing to consider, we have the different options so that we can test resources in arbitrary states with arbitrary mongo connectivity options, the panic here means we wouldn't be able to perform a test which ensures that attempting a connection to a non tls enabled MDB would fail. I don't think we'll ever need to test it, but the entire test pod itself will crash here rather than the specific test failing. What do you think?

Reply:
agree, I removed this check